rules/early-return/src/Rector/Return_/ReturnBinaryOrToEarlyReturnRector.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\EarlyReturn\Rector\Return_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\BooleanAnd;
+use PhpParser\Node\Expr\BinaryOp\BooleanOr;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\NodeManipulator\IfManipulator;
+use Rector\Core\PhpParser\Node\AssignAndBinaryMap;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\EarlyReturn\Tests\Rector\Return_\ReturnBinaryOrToEarlyReturnRector\ReturnBinaryOrToEarlyReturnRectorTest
+ */
+final class ReturnBinaryOrToEarlyReturnRector extends AbstractRector
+{
+    /**
+     * @var IfManipulator
+     */
+    private $ifManipulator;
+
+    /**
+     * @var AssignAndBinaryMap
+     */
+    private $assignAndBinaryMap;
+
+    public function __construct(IfManipulator $ifManipulator, AssignAndBinaryMap $assignAndBinaryMap)
+    {
+        $this->ifManipulator = $ifManipulator;
+        $this->assignAndBinaryMap = $assignAndBinaryMap;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Changes Single return of || to early returns', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function accept($something, $somethingelse)
+    {
+        return $something || $somethingelse;
```

Comment:
Notes from call - keep simple expressions:

```php
// keep
return 1 || 2 || 5;


if (true && $value) {
}
```

Update method calls and static calls only:

```php
// changed
return $this->process(...) || $this->someMethod(...) || some::statiCall() || true;
```

Reply:
implemented.